OpenTap.Metrics/MetricManager.cs

```diff
@@ -221,8 +236,32 @@ public static MetricInfo CreatePollMetric<T>(IAdditionalMetricSources owner, Fun
         return CreateMetric<T>(owner, name, groupName, MetricKind.Poll, pollFunction);
     }
 
+    public static MetricInfo CreateNullablePollMetric<T>(IAdditionalMetricSources owner, Func<T> pollFunction, string name, string groupName)
+    {
+        if (pollFunction is null)
+            throw new ArgumentNullException(nameof(pollFunction));
+
+        EnsureNullableMetricIsValid<T>();
+        return CreateMetric(owner, name, groupName, MetricKind.Poll, pollFunction);
+    }
+
     public static MetricInfo CreatePushMetric<T>(IAdditionalMetricSources owner, string name, string groupName) where T : IConvertible
     {
         return CreateMetric<T>(owner, name, groupName, MetricKind.Push, null);
     }
+
+    public static MetricInfo CreateNullablePushMetric<T>(IAdditionalMetricSources owner, string name, string groupName)
+    {
+        EnsureNullableMetricIsValid<T>();
+        return CreateMetric<T>(owner, name, groupName, MetricKind.Push, null);
+    }
+
+    private static void EnsureNullableMetricIsValid<T>()
+    {
+        var underlyingType = Nullable.GetUnderlyingType(typeof(T));
+        if (underlyingType.GetInterface(nameof(IConvertible)) is null)
+        {
+            throw new InvalidOperationException($"Unsupported type '{typeof(T)}' is not a 'System.IConvertible' type");
+        }
+    }
```

Comment:
Let's remove the IConvertible type restriction

Reply:
Done